### DISCUSSION TITLE
[Question] Where can I find the implementation of the GlowFilter in Ruffle/FlashSpec?

### DISCUSSION THREAD

**Discussion by TheIceCreamBear** [ruffle-rs/ruffle#10702] [General] [answered]
Hello all,

I am attempting to convert a Flash-based UI system to be used in a different engine. While exploring the flash files using [JPEXS](https://github.com/jindrapetrik/jpexs-decompiler), I have come across multiple Text objects that have a GlowFilter in their filter list. I have not been able to find a good explanation of how this was implemented.

My question to the Ruffle community is this:
- Where can I find the math and code behind *applying* a GlowFilter in Ruffle?
- Where can I find a good specification on the math for a GlowFilter?
   - (I have looked at [this spec](https://www.mobilefish.com/download/flash/swf_file_format_spec_v9.pdf) and found where it talks about GlowFilter, but it does not contain information that clearly defines how it works)
   
Thank you

**Comment by Lord-McSweeney** [comment]
Ruffle does not yet support `GlowFilter` at all.

**Reply by TheIceCreamBear** [reply]
Do you have any good public documentation I can look at to understand how it works then?

**Reply by coder0107git** [reply]
When I [Googled](https://www.google.com/search?q=actionscript+glowfilter) the documentation, I found [this](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/filters/GlowFilter.html) and [this](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/spark/filters/GlowFilter.html).

**Reply by TheIceCreamBear** [reply, marked as answer]
> When I [Googled](https://www.google.com/search?q=actionscript+glowfilter) the documentation, I found [this](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/flash/filters/GlowFilter.html) and [this](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/spark/filters/GlowFilter.html).

Unfortunately the linked documentation is how to _use_ the filter in flash or spark, as opposed to what I am looking of how it was _implemented_ under the hood of the rendering engine (which is why I initially came to ruffle hopping they had it implemented and readable).

However using I was able to google fu my way into finding OpenFL, which has an [implementation of the glow filter](https://github.com/openfl/openfl/blob/develop/src/openfl/filters/GlowFilter.hx) that I will be looking at later for inspiration.
